a3p-integration/proposals/z:acceptance/auction.test.js

```diff
@@ -0,0 +1,220 @@
+ * - Push price so that 1 ATOM is 50 ISTs
+ * - Wait until the auctioneer captures the price we just pushed
+ * - Fund actors
+ *  - gov1 gets 100 ATOMs
+ *  - user1 gets 90 ISTs
+ *  - gov3 gets 150 ISTs
+ * - Place bids for user1 and gov3 following the values in "config"
+ * - Deposit 100 ATOMs into book0, gov1 is the depositor
+ * - Wait until placed bids get their payouts
+ * - Wait until proceeds are distributed to the depositor
+ * - Make sure all actors receive the correct payouts
+ */
+
+/** @typedef {import('./test-lib/sync-tools.js').RetryOptions} RetryOptions */
+
+import {
+  agd,
+  agoric,
+  getUser,
+  GOV1ADDR,
+  GOV3ADDR,
+  USER1ADDR,
+} from '@agoric/synthetic-chain';
+import '@endo/init';
+import test from 'ava';
+import { boardSlottingMarshaller, makeFromBoard } from './test-lib/rpc.js';
+import { retryUntilCondition } from './test-lib/sync-tools.js';
+import {
+  calculateRetryUntilNextStartTime,
+  checkBidsOutcome,
+  checkDepositOutcome,
+  checkPriceCaptured,
+  depositCollateral,
+  fundAccts,
+  placeBids,
+  pushPricesForAuction,
+  scale6,
+} from './test-lib/auction-lib.js';
+
+const ambientAuthority = {
+  query: agd.query,
+  follow: agoric.follow,
+  setTimeout,
+};
+
+const fromBoard = makeFromBoard();
+const marshaller = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+
+const config = {
+  depositor: {
+    name: 'gov1',
+    addr: GOV1ADDR,
+    depositValue: '100000000',
+    offerId: `gov1-deposit-${Date.now()}`,
+  },
+  price: 50.0,
+  longLivingBidSetup: {
+    name: 'long-living-bidder',
+    // This bid is placed in an earlier proposal
+    give: '80IST',
+  },
+  currentBidsSetup: {
+    user1: {
+      bidder: USER1ADDR,
+      bidderFund: {
+        value: 90000000,
+        denom: 'uist',
+      },
+      offerId: `user1-bid-${Date.now()}`,
+      give: '90IST',
+      price: 46,
+    },
+    gov3: {
+      bidder: GOV3ADDR,
+      bidderFund: {
+        value: 150000000,
+        denom: 'uist',
+      },
+      offerId: `gov3-bid-${Date.now()}`,
+      give: '150IST',
+      discount: '13',
+    },
+  },
+  bidsOutcome: {
+    longLivingBidder: {
+      payouts: {
+        Bid: 0,
+        Collateral: 1.68421,
+      },
+    },
+    user1: {
+      payouts: {
+        Bid: 0,
+        Collateral: 2.0,
+      },
+    },
+    gov3: {
+      payouts: {
+        Bid: 0,
+        Collateral: 3.448275,
+      },
+    },
+  },
+};
+
+test.before(async t => {
+  /** @type {RetryOptions} */
+  const pushPriceRetryOpts = {
+    maxRetries: 5, // arbitrary
+    retryIntervalMs: 5000, // in ms
+  };
+
+  /** @type {RetryOptions} */
+  const bankSendRetryOpts = {
+    maxRetries: 3, // arbitrary
+    retryIntervalMs: 3000, // in ms
+  };
+
+  // Get current round id
+  const round = await agoric.follow(
+    '-lF',
+    ':published.priceFeed.ATOM-USD_price_feed.latestRound',
+  );
+  t.context = {
+    roundId: parseInt(round.roundId, 10),
+    retryOpts: {
```

Comment:
please `harden` this object so it's clear the opts are not to be modified during the test

Reply:
Resolved in https://github.com/Agoric/agoric-sdk/pull/10229/commits/990126165e1c8ba2ac7cc8a52ad696cf7fc347a3